processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -1474,7 +1474,15 @@ public Rowboat apply(Rowboat lhs, Rowboat rhs)
       Object[] rhsMetrics = rhs.getMetrics();
 
       for (int i = 0; i < metrics.length; ++i) {
-        metrics[i] = metricAggs[i].combine(lhsMetrics[i], rhsMetrics[i]);
+        Object lhsMetric = lhsMetrics[i];
+        Object rhsMetric = rhsMetrics[i];
+        if (lhsMetric == null) {
+          metrics[i] = rhsMetric;
+        } else if (rhsMetric == null) {
+          metrics[i] = lhsMetric;
```

Comment:
If i understand correctly, for the "both null" case on a long/float metric, the metric value for that merged row would be serialized as 0, so an aggregator would see 0 on that row, whether the conversion was done here in apply() or at the serializer